### DISCUSSION TITLE
Has anyone encountered this issue in their testing of RV

### DISCUSSION THREAD

**Discussion by nicholas132000** [NUS-CG3207/labs#14] [Lab 2] [answered]
u introduce ur first Instr..
then the ALUResult is supposed to be written to this Rd, say R1

then u delay for a clock period

then u introduce ur new Instr
then the ALUrResult that was suppsoed to be written to r1 in the first instr gets written to the Rd of the second instr instead, say r2

**Comment by Rayleigh47** [comment]
I encountered something similar. In my case I tried running several load instructions:

li s1, 10
li s2, 20
li s3, 30

simulating RegFile with these inputs seem to be ok:

![image](https://github.com/user-attachments/assets/276fef4b-d80e-4967-90cc-22a902c634c4)

rd stores the index of the registers (x0, x18, x19) 
rs2 stores the immediate value (10, 20, 30)
WE is high during this period and the values are recorded into RegBank successfully

Simulating RV with the instructions, the signals to rd, rs2 and WE are the same:

![image](https://github.com/user-attachments/assets/81c67fcd-3510-4ac3-bb49-729db0b19e8f)

Now instead of getting (RegBank[rd] = rs2), I am getting (RegBank[rd] = OLD rs2)
i.e. x0 stores the value 0 (instead of 10), x18 stores the value 10 (instead of 20), x19 stores the value 20 (instead of 30).

**Reply by Rayleigh47** [reply]
I managed to solve this behaviour by aligning to @(posedge CLK) in testbench and checking signals @(negedge CLK).

**Reply by r-panicker** [reply, marked as answer]
I guess you're trying to simulate the RV module directly?  
The likely issue is indeed with trying to write and check at the same edge.
Perhaps you should simulate the wrapper via test_Wrapper with appropriate modifications to the stimuli depending on the assembly program behaviour. The Wrapper acts as a convenient testbed for simulating the RV module.

**Reply by nicholas132000** [reply]
i see, thank you prof and @Rayleigh47
